sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -911,7 +911,8 @@ class CastChecks extends ExprChecks {
   val calendarChecks: TypeSig = none
   val sparkCalendarSig: TypeSig = CALENDAR + STRING
 
-  val arrayChecks: TypeSig = none
+  val arrayChecks: TypeSig = ARRAY.nested(FLOAT + DOUBLE + INT + ARRAY)
```

Comment:
This is still not enough to stop someone from casting an Array[Byte] to an Array[Double]. This only is checking the to, if the from is an ArrayType of anything. We need something for the from checks on an array.

Reply:
added fallback tests for this.